pkg/engine/artifacts.go

```diff
@@ -36,133 +36,111 @@ func kubernetesContainerAddonSettingsInit(p *api.Properties) map[string]kubernet
 			sourceFile:      "kubernetesmasteraddons-heapster-deployment.yaml",
 			base64Data:      k.GetAddonScript(HeapsterAddonName),
 			destinationFile: "kube-heapster-deployment.yaml",
-			isEnabled:       k.IsAddonEnabled(HeapsterAddonName),
```

Comment:
Upside: we remove duplicate code!

Reply:
do we still need `isEnabled` to be part of `kubernetesComponentFileSpec` or is it removed everywhere?

Reply:
nevermind it's still used in `kubernetesAddonSettingsInit`. What's the difference with `kubernetesContainerAddonSettingsInit` and why remove it in one but not both?

Reply:
It's still being used by the non-user-configurable addons, and the manifests.

I'm scaffolding out a refactor plan to move all addons into the "user-configurable addons" interface, and to do something similar for the `/etc/k8s/manifests` files as well, after which time we can possibly remove the property altogether.

Reply:
The difference is that `kubernetesContainerAddonSettingsInit` is able to correlate with an `KubernetesAddon` item in the `Addons` array.

`kubernetesAddonSettingsInit`, on the other hand, *is* where the authoritative state of whether or not a particular "non-user-configurable addon" is enabled or not. (This is tech debt.)